lib/taurus/qt/qtgui/display/tauruslabel.py

```diff
@@ -90,6 +90,8 @@ def _updateForeground(self, label):
         elif fgRole.lower() in ('', 'none'):
             pass
         else:
+            label.modelFragmentName = fgRole
```

Comment:
I do not like the idea of altering the label.modelFragmentName every time that you call _updateForeground on its controller (this is a side-efect and it is asking for future trouble, IMHO).
At the very least, I think that the original fragment name should be resored just after calling resetFormat.

...and even that would look quite ugly.

Isn't there any more elegant solution?

Reply:
> Isn't there any more elegant solution?

I did not find it, the other solution is at the taurusform level calling  crl.update  method after set a new Format